### ISSUE TITLE
Contrasts between factor levels at different values of another numeric predictor

### ISSUE THREAD

**Issue by DominiqueMakowski** [rvlenth/emmeans#96]
Dear Russ,

Let's take this model as a silly example:
```r
model <- lm(Sepal.Width ~ Species * Petal.Width, data=iris)
```
Let's say I am interested in the pairwise contrasts between the species levels:

```r
model %>%
    emmeans::emmeans("Species", by="Petal.Width") %>%
    emmeans::contrast(method = "pairwise")
```
```
Petal.Width = 1.199333:
 contrast                estimate        SE  df t.ratio p.value
 setosa - versicolor    1.5895853 0.3935652 144   4.039  0.0003
 setosa - virginica     1.7740849 0.4132843 144   4.293  0.0001
 versicolor - virginica 0.1844996 0.1450009 144   1.272  0.4131
```

I would like to know if it was possible to estimate the contrasts at different points of the numeric predictor, for instance Petal.Width = c(1, 2)?
I've tried something like:
```r
model %>%
    emmeans::emmeans("Species", by=list("Petal.Width"=c(1, 2))) %>%
    emmeans::contrast(method = "pairwise")
```
But it didn't work. Is something like this possible? Thanks a lot!

**Comment by rvlenth** [comment]
Use `at`. See `? ref_grid` and the "basics" vignette on [altering the reference grid](https://cran.r-project.org/web/packages/emmeans/vignettes/basics.html#altering)

**Comment by DominiqueMakowski** [comment]
I've missed that, so sorry! thanks again

**Comment by DominiqueMakowski** [comment]
Dear @rvlenth, sorry to bother again. I've tried playing around with the ref_grid and exploring the vignette you mentioned, but could not manage to obtain what I want, i.e., pairwise contrasts between levels of one factor within each value of a numeric variable. Something that would give a return like this:

```r
model <- lm(Sepal.Width ~ Species * Petal.Width, data=iris)
emmeans(model, someparameter=list(Petal.Width = c(0, 1, 2))
```
```
       contrast               estimate    SE  df t.ratio p.value
 0, setosa -  0, versicolor       1.590 0.394 144 4.039   0.0003 
 0, setosa -  0, virginica        1.774 0.413 144 4.293   0.0001 
 0, versicolor -  0, virginica    0.184 0.145 144 1.272   0.4131 

       contrast               estimate    SE  df t.ratio p.value
 1, setosa -  1, versicolor       1.590 0.394 144 4.039   0.0003 
 1, setosa -  1, virginica        1.774 0.413 144 4.293   0.0001 
 1, versicolor -  1, virginica    0.184 0.145 144 1.272   0.4131 

      contrast               estimate    SE  df t.ratio p.value
2, setosa - 2, versicolor       1.590 0.394 144 4.039   0.0003 
2, setosa - 2, virginica        1.774 0.413 144 4.293   0.0001 
2, versicolor - 2, virginica    0.184 0.145 144 1.272   0.4131 
```

With Daniel, we arrived to [the conclusion](https://github.com/easystats/estimate/issues/4) that the simplest way of having this output seems to be:

```r
model %>%
      emmeans::ref_grid(at = list(Petal.Width = c(0, 1, 2))) %>%
      emmeans::contrast(method = "pairwise") 
```
and then filter out all unwanted contrasts manually...  Is this the most straightforward approach? Thanks again!

**Comment by rvlenth** [comment]
Use ‘by’ like you would with any factor.

Sent from my iPhone

On Mar 13, 2019, at 5:35 AM, Dominique Makowski <notifications@github.com<mailto:notifications@github.com>> wrote:


Dear @rvlenth<https://github.com/rvlenth>, sorry to bother again. I've tried playing around with the ref_grid and exploring the vignette you mentioned, but could not manage to obtain what I want, i.e., pairwise contrasts between levels of one factor within each value of a numeric variable. Something that would give a return like this:

model <- lm(Sepal.Width ~ Species * Petal.Width, data=iris)
emmeans(model, someparameter=list(Petal.Width = c(0, 1, 2))

       contrast               estimate    SE  df t.ratio p.value
 0, setosa -  0, versicolor       1.590 0.394 144 4.039   0.0003
 0, setosa -  0, virginica        1.774 0.413 144 4.293   0.0001
 0, versicolor -  0, virginica    0.184 0.145 144 1.272   0.4131

       contrast               estimate    SE  df t.ratio p.value
 1, setosa -  1, versicolor       1.590 0.394 144 4.039   0.0003
 1, setosa -  1, virginica        1.774 0.413 144 4.293   0.0001
 1, versicolor -  1, virginica    0.184 0.145 144 1.272   0.4131

      contrast               estimate    SE  df t.ratio p.value
2, setosa - 2, versicolor       1.590 0.394 144 4.039   0.0003
2, setosa - 2, virginica        1.774 0.413 144 4.293   0.0001
2, versicolor - 2, virginica    0.184 0.145 144 1.272   0.4131


With Daniel, we arrived to the conclusion<https://github.com/easystats/estimate/issues/4> that the simplest way of having this output seems to be:

model %>%
      emmeans::ref_grid(at = list(Petal.Width = c(0, 1, 2))) %>%
      emmeans::contrast(method = "pairwise")

and then filter out all unwanted contrasts manually... Is this the most straightforward approach? Thanks again!
